tests/integration_tests/performance/test_network_latency.py

```diff
@@ -113,7 +114,10 @@ def test_network_latency(
 
     Send a ping from the guest to the host.
     """
-    requests = 1000
+
+    # each iteration is 6 seconds
+    # * 30 iterations = 5 minutes
+    requests = 30
```

Comment:
Instead of doing this, can we instead "backdate" the datapoints that we publish to cloudwatch? E.g. have the consumer return a 4-tuple that also include a timestamp? It seems that from a quick glance at the aws-embedded-metrics source, this should be possible by manually updating `metrics.meta['TimeStamp'] = ...`.

Reply:
This will also be easier to adapt for the remaining perf tests

Reply:
It may be possible, but it may be using the API in a way that is not documented.

What would it buy us, beyond being able do the metrics outside of the "stats" code?

One downside if the test breaks we would lose all the datapoints, though this doesn't happen often.

Reply:
The underlying put-metrics-data seems to explicitly support backdating the metrics: https://docs.aws.amazon.com/cli/latest/reference/cloudwatch/put-metric-data.html.

If we don't backdate the timestamps, we'll just end up with a lot of data points at the same time, which will not really allow us to analyse/plot them usefully. And some of our other tests cannot really be split apart because they already only collect ~20 samples (vsock/tcp throughput for example). Reducing this even more would degrade the value of the data we collect (even if we allow each pass its own warm-up period, which does not seem like a good use of resources because we'd be spending more time warming up than actually collecting performance data). So for these, we'd just have to increase the overall duration, but at this point it makes more sense to me to shard the test across multiple instances, and have each pass happen on its own .metal instance to account for hardware variability. 

I suppose splitting this test, and assigning proper timestamps aren't mutually exclusive, but I don't really see why we _wouldn't_ want properly timestamped data. And for other test, I don't see how this splitting would work :(

Reply:
I've synced with Pablo offline, it seems that even if we backdate the timestamps, as long as they are high-frequency, cloudwatch will aggregate at the log level already (and combine everything at 1 second granularity). It does keep ordering though, so there's not really a point in doing this at this stage (maybe for other tests, but lets experiment with the output of this test for a bit first). Anything else can definitely be done later :)